src/class/midi/midi_host.c

```diff
@@ -222,11 +229,20 @@ bool midih_open(uint8_t rhport, uint8_t dev_addr, tusb_desc_interface_t const *d
 
     p_desc = tu_desc_next(p_desc);
     desc_itf = (const tusb_desc_interface_t *)p_desc;
-    TU_VERIFY(TUSB_CLASS_AUDIO == desc_itf->bInterfaceClass);
     p_midi->itf_count = 1;
-  }
+    // See issue #3159
+    while ((p_desc < p_end) && (tu_desc_next(p_desc) <= p_end) && (desc_itf->bDescriptorType != TUSB_DESC_INTERFACE || (desc_itf->bInterfaceClass == TUSB_CLASS_AUDIO && desc_itf->bInterfaceSubClass != AUDIO_SUBCLASS_MIDI_STREAMING)))
```

Comment:
are we skipping until we got TUSB_DESC_INTERFACE/TUSB_DESC_INTERFACE/AUDIO_SUBCLASS_MIDI_STREAMING , can we just use 

```
!(desc_itf->bDescriptorType == TUSB_DESC_INTERFACE && desc_itf->bInterfaceClass == TUSB_CLASS_AUDIO && desc_itf->bInterfaceSubClass == AUDIO_SUBCLASS_MIDI_STREAMING)
```

Reply:
ah right, sorry, I was staying up late.